prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -449,6 +450,70 @@ ssize_t rxr_pkt_wait_handshake(struct rxr_ep *ep, fi_addr_t addr, struct rxr_pee
 	return 0;
 }
 
+static inline
+void rxr_pkt_handle_data_copied(struct rxr_ep *ep,
+				struct rxr_pkt_entry *pkt_entry)
+{
+	int pkt_type;
+
+	pkt_type = rxr_get_base_hdr(pkt_entry->pkt)->type;
+
+	switch(pkt_type) {
+	case RXR_DATA_PKT:
+	case RXR_READRSP_PKT:
+		rxr_data_pkt_handle_data_copied(ep, pkt_entry);
+		break;
+	case RXR_EAGER_MSGRTM_PKT:
+	case RXR_EAGER_TAGRTM_PKT:
+		rxr_pkt_handle_eager_rtm_data_copied(ep, pkt_entry);
```

Comment:
Splitting out all the rtm data copied cases feels ugly.  Also, still feels like we have the state machine steps wrong here and are treating this as a special state machine off to the side.

Reply:
Yes, you are right. I was mixed "data received" state and "data copied" state. CTS packet only depends on data being received. writing completion and releasing rx entry depend on data being copied. 
The new revision split the two state, therefore this function ```rxr_pkt_handle_data_copied``` is much simpler.